doc/lib/provisioning.txt

```diff
@@ -47,25 +47,34 @@ In addition to the components above, the Provisioning library also depends on C
 @page provisioning_design Design
 @brief Provisioning library architecture.
 
-The Provisioning library uses MQTT publish/subscribe to interact with the AWS IoT Core service for the Fleet Provisioning feature. Each library operation ("CreateKeysAndCertificate", "RegisterThing") uses two MQTT subscriptions, one for the accepted topic and one for the rejected topic, to obtain the AWS Iot Core response to the MQTT publish request that the operation makes. Each operation also requires a user-callback for handling the server response. The callbacks will run in MQTT's taskpool context.
+The Provisioning library uses MQTT publish/subscribe to interact with the AWS IoT Core service for the Fleet Provisioning feature. Each library operation ("CreateKeysAndCertificate", "CreateCertificateFromCsr", "RegisterThing") uses two MQTT subscriptions, one for the "accepted" topic and one for the "rejected" topic, to obtain the responses from AWS IoT Core, and publishes to the MQTT request topic of the operation. Each operation also requires a user-callback for handling the server response. The callbacks will run in MQTT's taskpool context.
```

Comment:
Is the taskpool context going to be owned by MQTT? Isn't it a system wide taskpool in current v4_beta based C-SDK?

Reply:
Good point. I had re-used the text from [Jobs' documentation](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta/doc/lib/jobs.txt#L52) which uses the phrase. 
I think the original intent of the taskpool was that it would be instantiated  by libraries for them to have their individual worker threads. However, for MCU optimization, the "system taskpool" change was made for sharing the same service across all libraries.

I will update the text to mention "_system_ taskpool context"